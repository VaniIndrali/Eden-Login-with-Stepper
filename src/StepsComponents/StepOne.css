.mainText {
	margin-top: 3.125rem;
}

.subText {
	color: grey;
	margin-bottom: 1.875rem;
}

.btn-bd-primary {
	font-weight: 600;
	color: white;
	background-color: #744ab7;
	border-color: #744ab7;
	border-radius: 0.25rem;
	border: 0;
	padding: 0.625rem 7.938rem;
	align-items: center;
}

.Connector-d3-0-2-19 {
	border-image: linear-gradient(
			45deg,
			#744ab7 0 50%,
			rgb(189, 189, 189) 50% 100%
		)
		1;
}
