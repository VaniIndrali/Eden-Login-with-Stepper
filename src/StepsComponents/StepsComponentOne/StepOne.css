.Connector-d3-0-2-19:not(.active):not(.completed) {
	border-image: linear-gradient(
			45deg,
			#744ab7 0 50%,
			rgb(189, 189, 189) 50% 100%
		)
		1;
}

.Connector-d3-0-2-19.active {
	border-color: #744ab7 !important;
}

.Connector-d3-0-2-19.completed {
	border-color: #744ab7 !important;
}
