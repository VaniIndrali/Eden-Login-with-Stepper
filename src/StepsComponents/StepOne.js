import { React, useState } from "react";
import "./StepOne.css";

export const StepOne = (props) => {
	const [displayName, setDisplayName] = useState("");

	const handleChange = (e) => {
		setDisplayName(e.target.value);
	};
	const handleSubmit = (e) => {
		e.preventDefault();
		props.onSubmit(displayName);
	};

	return (
		<div id="login">
			<div className="container">
				<h2 className="text-center mainText">
					Welcome! First things first...
				</h2>
				<h6 className="text-center subText">
					You can always change them later.
				</h6>
				<div
					id="login-row"
					className="row justify-content-center align-items-center"
				>
					<div
						id="login-column"
						className="col-md-11"
					>
						<div
							id="login-box"
							className="col-md-12"
						>
							<form
								id="login-form"
								className="form"
								onSubmit={(e) => handleSubmit(e)}
							>
								<div className="form-group">
									<label
										htmlFor="fullname"
										className="text-left"
									>
										Full Name
									</label>
									<br />
									<input
										type="text"
										name="fullname"
										id="fullname"
										className="form-control"
										placeholder="Steve Jobs"
									/>
								</div>
								<div className="form-group">
									<label
										htmlFor="displayname"
										className="text-left"
									>
										Display Name
									</label>
									<br />
									<input
										type="text"
										value={displayName}
										name="displayname"
										id="displayname"
										className="form-control"
										placeholder="Steve"
										onChange={(e) => handleChange(e)}
									/>
								</div>
								<div className="form-group">
									<input
										type="submit"
										name="Create Workspace"
										className="btn-bd-primary"
										value="Create Workspace"
									/>
								</div>
							</form>
						</div>
					</div>
				</div>
			</div>
		</div>
	);
};

export default StepOne;
