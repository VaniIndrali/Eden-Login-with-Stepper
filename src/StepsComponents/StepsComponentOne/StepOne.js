import { React, useState } from "react";
import "../StepsComponentCss/StepsCommon.css";
import "./StepOne.css";

export const StepOne = (props) => {
	const [displayName, setDisplayName] = useState("");

	const handleChange = (e) => {
		setDisplayName(e.target.value);
	};
	const handleSubmit = (e) => {
		e.preventDefault();
		props.onSubmit(displayName);
	};

	return (
		<div id="loginOne d-flex justify-content-center">
			<div className="container col-md-12 col-9">
				<h6 className="text-center subText">
					You can always change them later.
				</h6>
				<div
					id="login-row"
					className="row justify-content-center align-items-center"
				>
					<div
						id="login-column"
						className="col-md-11"
					>
						<div
							id="login-box"
							className="col-md-12"
						>
							<form
								id="login-form"
								className="form"
								onSubmit={(e) => handleSubmit(e)}
							>
								<div className="form-group">
									<label
										htmlFor="fullname"
										className="text-left"
									>
										Full Name
									</label>
									<br />
									<input
										type="text"
										name="fullname"
										id="fullname"
										className="form-control"
										placeholder="Steve Jobs"
										required
									/>
								</div>
								<div className="form-group">
									<label
										htmlFor="displayname"
										className="text-left"
									>
										Display Name
									</label>
									<br />
									<input
										type="text"
										value={displayName}
										name="displayname"
										id="displayname"
										className="form-control"
										placeholder="Steve"
										required
										maxlength="11"
										onChange={(e) => handleChange(e)}
									/>
								</div>
								<div className="form-group">
									<input
										type="submit"
										name="Create Workspace"
										className="button"
										value="Create Workspace"
									/>
								</div>
							</form>
						</div>
					</div>
				</div>
			</div>
		</div>
	);
};

export default StepOne;
