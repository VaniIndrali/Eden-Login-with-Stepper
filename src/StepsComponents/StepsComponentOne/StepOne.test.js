import { render, screen } from "@testing-library/react";
import StepOne from "./StepOne";

test("renders StepOne", () => {
	const props = {
		onSubmit: jest.fn(),
	};

	const component = render(<StepOne {...props} />);

	const subHeading = screen.getByRole("heading");
	expect(subHeading.textContent).toEqual("You can always change them later.");

	expect(screen.getByText("Full Name")).toBeTruthy();
	expect(screen.getByText("Display Name")).toBeTruthy();
	expect(
		screen.getByRole("textbox", {
			name: "Full Name",
		})
	).toBeTruthy();
	expect(
		screen.getByRole("textbox", {
			name: "Display Name",
		})
	).toBeTruthy();
	expect(
		screen.getByRole("button", {
			name: "Create Workspace",
		})
	).toBeTruthy();

	expect(component).toMatchSnapshot();
});
