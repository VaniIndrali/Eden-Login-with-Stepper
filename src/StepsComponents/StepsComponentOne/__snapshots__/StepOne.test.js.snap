// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StepOne 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="loginOne d-flex justify-content-center"
      >
        <div
          class="container col-md-12 col-9"
        >
          <h6
            class="text-center subText"
          >
            You can always change them later.
          </h6>
          <div
            class="row justify-content-center align-items-center"
            id="login-row"
          >
            <div
              class="col-md-11"
              id="login-column"
            >
              <div
                class="col-md-12"
                id="login-box"
              >
                <form
                  class="form"
                  id="login-form"
                >
                  <div
                    class="form-group"
                  >
                    <label
                      class="text-left"
                      for="fullname"
                    >
                      Full Name
                    </label>
                    <br />
                    <input
                      class="form-control"
                      id="fullname"
                      name="fullname"
                      placeholder="Steve Jobs"
                      required=""
                      type="text"
                    />
                  </div>
                  <div
                    class="form-group"
                  >
                    <label
                      class="text-left"
                      for="displayname"
                    >
                      Display Name
                    </label>
                    <br />
                    <input
                      class="form-control"
                      id="displayname"
                      maxlength="11"
                      name="displayname"
                      placeholder="Steve"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="form-group"
                  >
                    <input
                      class="button"
                      name="Create Workspace"
                      type="submit"
                      value="Create Workspace"
                    />
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="loginOne d-flex justify-content-center"
    >
      <div
        class="container col-md-12 col-9"
      >
        <h6
          class="text-center subText"
        >
          You can always change them later.
        </h6>
        <div
          class="row justify-content-center align-items-center"
          id="login-row"
        >
          <div
            class="col-md-11"
            id="login-column"
          >
            <div
              class="col-md-12"
              id="login-box"
            >
              <form
                class="form"
                id="login-form"
              >
                <div
                  class="form-group"
                >
                  <label
                    class="text-left"
                    for="fullname"
                  >
                    Full Name
                  </label>
                  <br />
                  <input
                    class="form-control"
                    id="fullname"
                    name="fullname"
                    placeholder="Steve Jobs"
                    required=""
                    type="text"
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class="text-left"
                    for="displayname"
                  >
                    Display Name
                  </label>
                  <br />
                  <input
                    class="form-control"
                    id="displayname"
                    maxlength="11"
                    name="displayname"
                    placeholder="Steve"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="form-group"
                >
                  <input
                    class="button"
                    name="Create Workspace"
                    type="submit"
                    value="Create Workspace"
                  />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
